src/debugging/DebugConfigCreator.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and others.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as fs from 'fs';
+
+import { WorkspaceFolder } from 'vscode';
+import { TaskCreator } from './TaskCreator';
+import { LaunchConfigCreator } from './LaunchConfigCreator';
+import { getQuarkusProjectBuildSupport } from '../utils/workspaceUtils';
+import { IBuildSupport } from '../definitions/IBuildSupport';
+
+/**
+ * This class is responsible for creating the .vscode/ folder (if it does not
+ * exist already) and adding the launch configuration and task that runs the
+ * Quarkus application in `workspaceFolder` when a debug session starts
+ */
+export class DebugConfigCreator {
+
+  public static debugConfigCreator: DebugConfigCreator;
+
+  private dotVSCodeDir: string;
+
+  public static async createFiles(workspaceFolder: WorkspaceFolder): Promise<void> {
+    DebugConfigCreator.debugConfigCreator = new DebugConfigCreator(workspaceFolder);
```

Comment:
avoid mutating static constants in static methods. You can prolly remove the static keyword from debugConfigCreator

Reply:
> avoid mutating static constants in static methods

Is this a rule/convention that is commonly followed?

Reply:
if 2 threads start mutating the same static object, you're in trouble